pkg/inspecttypes/dockercompat/dockercompat_test.go

```diff
@@ -226,12 +237,126 @@ func TestContainerFromNative(t *testing.T) {
 				Config: &Config{
 					Hostname: "host1",
 				},
+				NetworkSettings: &NetworkSettings{
+					Ports:    &nat.PortMap{},
+					Networks: map[string]*NetworkEndpointSettings{},
+				},
 			},
 		},
 	}
 
 	for _, tc := range testcase {
+		fmt.Fprintln(os.Stderr, tc.name)
 		d, _ := ContainerFromNative(tc.n)
 		assert.DeepEqual(t, d, tc.expected)
 	}
 }
+
+func TestNetworkSettingsFromNative(t *testing.T) {
+	tempStateDir, err := os.MkdirTemp(t.TempDir(), "rw")
+	if err != nil {
+		t.Fatal(err)
+	}
+	os.WriteFile(filepath.Join(tempStateDir, "resolv.conf"), []byte(""), 0644)
+	defer os.RemoveAll(tempStateDir)
+
+	testcase := []struct {
+		name     string
+		n        *native.NetNS
+		s        *specs.Spec
+		expected *NetworkSettings
+	}{
+		// Given null native.NetNS, Return initialized NetworkSettings
+		//    UseCase: Inspect a Stopped Container
+		{
+			name: "Given Null NetNS, Return initialized NetworkSettings",
+			n:    nil,
+			s:    &specs.Spec{},
+			expected: &NetworkSettings{
+				Ports:    &nat.PortMap{},
+				Networks: map[string]*NetworkEndpointSettings{},
+			},
+		},
+		// Given native.NetNS with single Interface with Port Annotations, Return populated NetworkSettings
+		//   UseCase: Inspect a Running Container with published ports
+		{
+			name: "Given NetNS with single Interface with Port Annotation, Return populated NetworkSettings",
+			n: &native.NetNS{
+				Interfaces: []native.NetInterface{
+					{
+						Interface: net.Interface{
+							Index: 1,
+							MTU:   1500,
+							Name:  "eth0.100",
+							Flags: net.FlagUp,
+						},
+						HardwareAddr: "xx:xx:xx:xx:xx:xx",
+						Flags:        []string{},
+						Addrs:        []string{"10.0.4.30/24"},
+					},
+				},
+			},
+			s: &specs.Spec{
+				Annotations: map[string]string{
+					"nerdctl/ports": "[{\"HostPort\":8075,\"ContainerPort\":77,\"Protocol\":\"tcp\",\"HostIP\":\"127.0.0.1\"}]",
+				},
+			},
+			expected: &NetworkSettings{
+				Ports: &nat.PortMap{
+					nat.Port("77/tcp"): []nat.PortBinding{
+						{
+							HostIP:   "127.0.0.1",
+							HostPort: "8075",
+						},
+					},
+				},
+				Networks: map[string]*NetworkEndpointSettings{
+					"unknown-eth0.100": {
+						IPAddress:   "10.0.4.30",
+						IPPrefixLen: 24,
+						MacAddress:  "xx:xx:xx:xx:xx:xx",
+					},
+				},
+			},
+		},
+		// Given native.NetNS with single Interface without Port Annotations, Return populated NetworkSettings
+		//   UseCase: Inspect a Running Container without published ports
+		{
+			name: "Given NetNS with single Interface without Port Annotations, Return populated NetworkSettings",
```

Comment:
nit: initialized network setting.

Reply:
This was a difficult comment and test case name to write.
Technically, a "Running Container without published ports" will still have a populated NetworkSettings entity that contains an "initialized Ports" child entity.

I will try to update the comment to reflect this statement, if this clarification makes sense to you.

Reply:
thanks

Reply:
@Shubhranshu153  within commit `adf4035`, I also updated the test description/name to reflect this conversation.